app/2.0/docs/tools/glossary.md

```diff
@@ -0,0 +1,160 @@
+
+### `polymer lint`
+Invokes [`polymer-linter`](#polymer-linter) to [lint](#linter) a polymer project.
+
+### `polymer serve`
+This command starts [`polyserve`](#polyserve), a simple development server for Web Components.
+
+### `polymer test`
+Runs tests for your Polymer app with [`web-component-tester`](#web-component-tester).
+
+## polymer-build
+`polymer-build` is a library for building Polymer projects. `polymer-build` allows you to completely customize your build and combine additional streams and build tasks in any order. 
+
+Compare the `polymer build` command, which calls `polymer-build` with preconfigured options. 
+
+## polymer-bundler
+Polymer build tool that follows HTML Imports and `<script>` tags to concatenate resources into a single page for production. Can improve performance by reducing network trips.
+
+## polymer-linter
+[Linter](#linter) invoked by `polymer lint` command. Lints a Polymer project.
+
+## polymer-analyzer
+A static analysis framework for web applications. 
+
+## `polymer.json`
+`polymer.json` is the Polymer build configuration file. It lets you configure your app’s entrypoint and app shell files, instead of passing these as parameters to polymer build.
+
+`polymer.json` also lets you specify the locations of source files, HTML fragments and additional dependencies, and configure the [Polymer linter](#polymer-linter).
+
+## `polyserve`
+`polyserve` is a locally-installable web server for developing [bower](#bower) components. 
+
+# R
+## routing
+Routing is the process an app uses to determine how to respond to a client request for a particular endpoint. A default route is the endpoint the app directs the user to when no route is specified. In a PWA, the [app shell](#app-shell) performs routing.
+
+# S
+## service worker
+A script that a web browser runs in the background. On first accessing a particular site, the service worker fetches a defined set of resources and caches them. Then, the service worker acts as a client-side proxy. It intercepts network requests for resources and, where possible, supplies the resources from the browser cache. 
+
+When a service worker is installed, apps may be able to be used offline in a meaningful way.
+
+## shell
+In the [app-shell](#app-shell) design pattern, the shell is the main code that boots the app and loads necessary resources. The shell usually includes the common UI for the app, and the [router](#routing), which loads resources needed for the current route.
+
+## static dependency
+A [dependency](#dependency) that is required by a given file, and whose import is hard-coded using syntax like `<link rel="import">` or `<script src="...">`. This dependency will always be loaded when the given file is loaded.
+
+## transitive dependencies
```

Comment:
Maybe make singular and say something like:

A transitive dependency is in indirect dependency, one that is depended on by a direct dependency or other indirect dependency.

The transitive dependencies of a resource are all dependencies, direct and indirect, that a resource depends on.